### PR TITLE
auto-deploy to production on merge to main

### PR DESCRIPTION
**Pull request by analogrelay**
Once merged, this should trigger a deployment to the new production resources on every commit to `main`.

### CODE REVIEW

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -79,6 +81,7 @@ jobs:
             --label "github.run_id=${{ github.run_id }}" \
             -f functions.Dockerfile \
             .
+            docker tag abbotimages.azurecr.io/aseriousbiz/abbot-py/runner-py-functions:$IMAGE_VERSION abbotimages.azurecr.io/aseriousbiz/abbot-py/runner-py-functions:$COMMIT_LABEL
```
Comment (dahlbyk) [inline .github/workflows/docker.yaml:84]: Bring over `env.IMAGE_BASE_NAME` for symmetry with JS workflow?

**Review by dahlbyk — APPROVED** [review]
Seems fine, harder to read than JS equivalent

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (analogrelay) [reply .github/workflows/docker.yaml:84]: It actually _had_ that variable originally, but this repo builds two images (`runner-py` and `runner-py-functions`) so I took it out. Eventually the other repos will too

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -66,6 +66,8 @@ jobs:
 
           # Save the IMAGE_VERSION env var to the GITHUB_ENV file so future steps can use it.
           echo "IMAGE_VERSION=$IMAGE_VERSION" >> $GITHUB_ENV
+          echo "COMMIT_LABEL=commit-${{ github.sha }}" >> $GITHUB_ENV
+          echo "COMMIT_TAG=${IMAGE_TAG_BASE}:commit-${{ github.sha }}" >> $GITHUB_ENV
```
Comment (dahlbyk) [inline .github/workflows/docker.yaml:70]: Where does `IMAGE_TAG_BASE` come from?

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (dahlbyk) [reply .github/workflows/docker.yaml:84]: Why not `IMAGE_BASE_NAME = abbotimages.azurecr.io/aseriousbiz/abbot-py`?

```suggestion
            docker tag $IMAGE_BASE_NAME/runner-py-functions:$IMAGE_VERSION $IMAGE_BASE_NAME/runner-py-functions:$COMMIT_LABEL
```

**Pull request merged**